uaclient/config.py

```diff
@@ -165,25 +169,73 @@ def https_proxy(self, value: str):
         self.write_cfg()
 
     @property
-    def apt_http_proxy(self) -> Optional[str]:
-        return self.cfg.get("ua_config", {}).get("apt_http_proxy")
+    def ua_apt_https_proxy(self) -> Optional[str]:
+        return self.cfg.get("ua_config", {}).get("ua_apt_https_proxy")
 
-    @apt_http_proxy.setter
-    def apt_http_proxy(self, value: str):
+    @ua_apt_https_proxy.setter
+    def ua_apt_https_proxy(self, value: str):
         if "ua_config" not in self.cfg:
             self.cfg["ua_config"] = {}
-        self.cfg["ua_config"]["apt_http_proxy"] = value
+        self.cfg["ua_config"]["ua_apt_https_proxy"] = value
         self.write_cfg()
 
     @property
-    def apt_https_proxy(self) -> Optional[str]:
-        return self.cfg.get("ua_config", {}).get("apt_https_proxy")
+    def ua_apt_http_proxy(self) -> Optional[str]:
+        return self.cfg.get("ua_config", {}).get("ua_apt_http_proxy")
 
-    @apt_https_proxy.setter
-    def apt_https_proxy(self, value: str):
+    @ua_apt_http_proxy.setter
+    def ua_apt_http_proxy(self, value: str):
         if "ua_config" not in self.cfg:
             self.cfg["ua_config"] = {}
-        self.cfg["ua_config"]["apt_https_proxy"] = value
+        self.cfg["ua_config"]["ua_apt_http_proxy"] = value
+        self.write_cfg()
+
+    @property
+    def global_apt_http_proxy(self) -> Optional[str]:
+        global_val = self.cfg.get("ua_config", {}).get("global_apt_http_proxy")
+        if global_val:
+            return global_val
+
+        old_apt_val = self.cfg.get("ua_config", {}).get("apt_http_proxy")
+        if old_apt_val:
+            event.info(
```

Comment:
Let's move that message into the `message` modules.

Also, I think this is a worth `event.warning` message for the json response case. What do you think @renanrodrigo @orndorffgrant ?

Reply:
I agree

Reply:
+1